web-runtime/npm-package/src/FaaS.ts

```diff
@@ -0,0 +1,173 @@
+import { WASI } from '@wasmer/wasi';
+import bindings from '@wasmer/wasi/lib/bindings/browser';
+import { WasmFs } from '@wasmer/wasmfs';
+
+import { init } from './marine_web_runtime';
+import { FaaSConfig } from './config';
+
+type LogLevel = 'info' | 'trace' | 'debug' | 'info' | 'warn' | 'error' | 'off';
+
+type LogImport = {
+    log_utf8_string: (level: any, target: any, offset: any, size: any) => void;
+};
+
+type ImportObject = {
+    host: LogImport;
+};
+
+type HostImportsConfig = {
+    exports: any;
+};
+
+const logFunction = (level: LogLevel, message: string) => {
+    switch (level) {
+        case 'error':
+            console.error(message);
+            break;
+        case 'warn':
+            console.warn(message);
+            break;
+        case 'info':
+            console.info(message);
+            break;
+        case 'debug':
+        case 'trace':
+            console.log(message);
+            break;
+    }
+};
+
+let cachegetUint8Memory0: any = null;
+
+function getUint8Memory0(wasm: any) {
+    if (cachegetUint8Memory0 === null || cachegetUint8Memory0.buffer !== wasm.memory.buffer) {
+        cachegetUint8Memory0 = new Uint8Array(wasm.memory.buffer);
+    }
+    return cachegetUint8Memory0;
+}
+
+function getStringFromWasm0(wasm: any, ptr: any, len: any) {
+    return decoder.decode(getUint8Memory0(wasm).subarray(ptr, ptr + len));
+}
+
+/// Returns import object that describes host functions called by AIR interpreter
+function newImportObject(cfg: HostImportsConfig): ImportObject {
+    return {
+        host: log_import(cfg),
+    };
+}
+
+function log_import(cfg: HostImportsConfig): LogImport {
+    return {
+        log_utf8_string: (level: any, target: any, offset: any, size: any) => {
+            let wasm = cfg.exports;
+
+            try {
+                let str = getStringFromWasm0(wasm, offset, size);
+                let levelStr: LogLevel;
+                switch (level) {
+                    case 1:
+                        levelStr = 'error';
+                        break;
+                    case 2:
+                        levelStr = 'warn';
+                        break;
+                    case 3:
+                        levelStr = 'info';
+                        break;
+                    case 4:
+                        levelStr = 'debug';
+                        break;
+                    case 6:
+                        levelStr = 'trace';
+                        break;
+                    default:
+                        return;
+                }
+                logFunction(levelStr, str);
+            } finally {
+            }
+        },
+    };
+}
+
+type Awaited<T> = T extends PromiseLike<infer U> ? U : T;
+
+type MarineInstance = Awaited<ReturnType<typeof init>> | 'not-set' | 'terminated';
+
+const decoder = new TextDecoder();
+
+export class FaaS {
+    private _controlModuleWasm: WebAssembly.Module;
```

Comment:
Let's call it also `_controlModule` or `_controlWasmModule`, because when I see `Wasm` at the end I think that it's actually a Wasm modules bytes.